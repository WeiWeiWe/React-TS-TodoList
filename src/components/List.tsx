import React from 'react';
import styled from 'styled-components';
import { Checkmark, PencilOutline, CloseOutline } from 'react-ionicons';

const ListStyle = styled.section`
  border-bottom: 1px solid #bbb;
  .list-fields {
    .list-item {
      display: flex;
      padding-top: 7px;
      .check-box {
        flex: 0 0 25px;
        height: 25px;
        margin: 2px 0 0 10px;
        border: 1px solid #bbb;

        .check-btn {
          display: none;
          cursor: pointer;
        }
        &:hover {
          .check-btn {
            display: block;
            color: rgba(50, 197, 250, 0.5);
          }
        }
      }
      input {
        flex: 1;
        margin-left: 15px;
        font-size: 18px;
        border: none;
        outline: none;
      }
      i {
        flex: 0 0 25px;
        cursor: pointer;
        .modify-btn {
          &:hover {
            color: rgba(50, 197, 250);
          }
        }
        .delete-btn {
          &:hover {
            color: rgba(255, 0, 0);
          }
        }
      }
    }
  }

  button {
    margin-left: 5px;
  }
`;

function List() {
  return (
    <ListStyle className="section-fields">
      <ul className="list-fields">
        <li className="list-item">
          <span className="check-box">
            <Checkmark cssClasses="check-btn" color={'rgba(50, 197, 250)'} />
          </span>
          <input type="text" />
          <i>
            <PencilOutline cssClasses="modify-btn" color={'rgba(50, 197, 250, 0.5)'} width="35px" height="30px" />
          </i>
          <i>
            <CloseOutline cssClasses="delete-btn" color={'rgba(255, 0, 0, 0.5)'} width="35px" height="30px" />
          </i>
        </li>
      </ul>
    </ListStyle>
  );
}

export default List;
