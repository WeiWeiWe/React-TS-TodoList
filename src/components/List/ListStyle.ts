import styled from 'styled-components';

export const ListStyle = styled.section`
  height: 300px;
  background: rgba(50, 197, 250);

  .list-fields {
    max-height: 290px;
    overflow: auto;

    .list-item {
      display: flex;
      align-items: center;
      padding-top: 7px;
      background: #fff;
      border-bottom: 1px solid #bbb;

      .list-item-check-box {
        flex: 0 0 25px;
        height: 25px;
        margin: 2px 0 0 10px;
        border: 1px solid #bbb;
        cursor: pointer;

        .check-box-btn {
          z-index: 1;
          display: none;
          cursor: pointer;
        }
        .check-box-btn.done {
          display: block;
        }
      }

      .list-item-input {
        flex: 1;
        margin-left: 15px;
        color: #000;
        font-size: 18px;
        background: #fff;
        border: none;
        outline: none;
      }
      .list-item-input.done {
        color: rgba(255, 0, 0, 0.9);
        text-decoration: line-through;
      }

      .list-item-modify-icon {
        flex: 0 0 25px;
        cursor: pointer;

        &:hover {
          .modify-btn {
            color: rgba(50, 197, 250, 0.5);
          }
        }

        .modify-btn.isClick {
          color: rgba(50, 197, 250, 0.5);
        }
      }

      .list-item-delete-icon {
        flex: 0 0 25px;
        cursor: pointer;

        &:hover {
          .delete-btn {
            color: rgba(255, 0, 0, 0.5);
          }
        }
      }
    }
    .list-item.prompt-tag {
      align-items: center;
      justify-content: center;

      h3 {
        color: rgba(50, 197, 250, 0.7);
        font-size: 24px;
      }
    }
  }
`;
