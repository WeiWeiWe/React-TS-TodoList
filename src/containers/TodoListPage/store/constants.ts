export const CHANGE_INPUT_VALUE = 'CHANGE_INPUT_VALUE';
export const ADD_TODO_LIST = 'ADD_TODO_LIST';
export const DELETE_TODO_LIST = 'DELETE_TODO_LIST';
export const CHANGE_LIST_ITEM_TITLE_VALUE = 'CHANGE_LIST_ITEM_TITLE_VALUE';
export const CHANGE_PROCESS_STATUS = 'CHANGE_PROCESS_STATUS';
export const CHANGE_TAB_STATUS = 'CHANGE_TAB_STATUS';
