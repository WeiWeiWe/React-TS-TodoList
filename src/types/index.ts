export * from './Type';
export * from './Enum';
