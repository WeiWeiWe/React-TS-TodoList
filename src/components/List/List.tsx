import React from 'react';
import { ListStyle } from './ListStyle';
import { Checkmark, PencilOutline, CloseOutline } from 'react-ionicons';
import { ItemTypes, InputArrayRefType, EnumItemProcessTypes } from '../../types';

export interface ListIprops {
  list: ItemTypes[];
  deleteTodoList: (item: ItemTypes) => void;
  switchModifyStatus: (item: ItemTypes) => void;
  InputBlur: (item: ItemTypes) => void;
  changeTitleInputValue: (e: React.ChangeEvent<HTMLInputElement>, item: ItemTypes) => void;
  keyUpTitleInputValueDone: (e: React.KeyboardEvent<HTMLInputElement>) => void;
  inputArrayRef: { current: InputArrayRefType[] };
  changeProcessStatus: (item: ItemTypes) => void;
  tabStatus: EnumItemProcessTypes;
}

export function List({
  list,
  deleteTodoList,
  switchModifyStatus,
  InputBlur,
  changeTitleInputValue,
  inputArrayRef,
  keyUpTitleInputValueDone,
  changeProcessStatus,
  tabStatus,
}: ListIprops) {
  const filterTabList =
    list?.length > 0
      ? list.filter((item) => {
          if (tabStatus === EnumItemProcessTypes.All) {
            return item;
          } else {
            return item.process === tabStatus;
          }
        })
      : [];

  const promptTag = () => {
    if (list?.length > 0) {
      if (tabStatus === EnumItemProcessTypes.DONE) {
        return <h3>No Done Items</h3>;
      } else if (tabStatus === EnumItemProcessTypes.ACTIVE) {
        return <h3>No Active Items</h3>;
      } else {
        return <h3>No Items</h3>;
      }
    } else {
      return <h3>No Items</h3>;
    }
  };

  return (
    <ListStyle>
      <ul className="list-fields">
        {filterTabList?.length > 0 ? (
          filterTabList.map((item, index) => {
            return (
              <li className="global-block-fields list-item" key={item.id} data-testid="list-item">
                <span
                  className="list-item-check-box"
                  onClick={() => changeProcessStatus(item)}
                  data-testid="list-item-check-box"
                >
                  <Checkmark
                    cssClasses={item.process === EnumItemProcessTypes.DONE ? 'check-box-btn done' : 'check-box-btn'}
                    color={'rgb(50, 197, 250)'}
                  />
                </span>
                <input
                  className={item.process === EnumItemProcessTypes.DONE ? 'list-item-input done' : 'list-item-input'}
                  data-testid="list-item-input"
                  disabled
                  type="text"
                  value={item.title}
                  onChange={(e: React.ChangeEvent<HTMLInputElement>) => changeTitleInputValue(e, item)}
                  onKeyUp={keyUpTitleInputValueDone}
                  onBlur={() => InputBlur(item)}
                  ref={(element) => {
                    if (element) {
                      inputArrayRef.current[index] = {
                        id: item.id,
                        element,
                      };
                    }
                  }}
                />
                <i className="list-item-modify-icon" onClick={() => switchModifyStatus(item)} data-testid="modify-btn">
                  <PencilOutline
                    cssClasses={item.isModify ? 'modify-btn isClick' : 'modify-btn'}
                    color={'rgb(50, 197, 250)'}
                    width="35px"
                    height="30px"
                  />
                </i>
                <i className="list-item-delete-icon" onClick={() => deleteTodoList(item)} data-testid="delete-btn">
                  <CloseOutline cssClasses="delete-btn" color={'rgba(255, 0, 0, 0.9)'} width="35px" height="30px" />
                </i>
              </li>
            );
          })
        ) : (
          <li className="global-block-fields list-item prompt-tag" data-testid="prompt-tag">
            {promptTag()}
          </li>
        )}
      </ul>
    </ListStyle>
  );
}
