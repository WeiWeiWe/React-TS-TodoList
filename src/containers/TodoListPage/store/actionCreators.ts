import * as constants from './constants';
import { ItemTypes, EnumItemProcessTypes } from '../../../types';

interface ChangeInputValueAction {
  type: typeof constants.CHANGE_INPUT_VALUE;
  value: string;
}

interface AddTodoListAction {
  type: typeof constants.ADD_TODO_LIST;
  item: ItemTypes;
}

interface DeleteTodoListAction {
  type: typeof constants.DELETE_TODO_LIST;
  id: string;
}

interface SwitchModifyStatusAction {
  type: typeof constants.SWITCH_MODIFY_STATUS;
  status: boolean;
  id: string;
}

interface ChangeListItemTitleValueAction {
  type: typeof constants.CHANGE_LIST_ITEM_TITLE_VALUE;
  value: string;
  id: string;
}

interface ChangeProcessStatusAction {
  type: typeof constants.CHANGE_PROCESS_STATUS;
  process: string;
  id: string;
}

interface ChangeTabStatusAction {
  type: typeof constants.CHANGE_TAB_STATUS;
  status: EnumItemProcessTypes;
}

export type TodoListActionTypes =
  | ChangeInputValueAction
  | AddTodoListAction
  | DeleteTodoListAction
  | SwitchModifyStatusAction
  | ChangeListItemTitleValueAction
  | ChangeProcessStatusAction
  | ChangeTabStatusAction;

export const changeInputValueActionCreator = (value: string): ChangeInputValueAction => {
  return {
    type: constants.CHANGE_INPUT_VALUE,
    value,
  };
};

export const addTodoListActionCreator = (item: ItemTypes): AddTodoListAction => {
  return {
    type: constants.ADD_TODO_LIST,
    item,
  };
};

export const deleteTodoListActionCreator = (id: string): DeleteTodoListAction => {
  return {
    type: constants.DELETE_TODO_LIST,
    id,
  };
};

export const switchModifyStatusActionCreator = (status: boolean, id: string): SwitchModifyStatusAction => {
  return {
    type: constants.SWITCH_MODIFY_STATUS,
    status,
    id,
  };
};

export const ChangeListItemTitleValueActionCreator = (value: string, id: string): ChangeListItemTitleValueAction => {
  return {
    type: constants.CHANGE_LIST_ITEM_TITLE_VALUE,
    value,
    id,
  };
};

export const ChangeProcessStatusActionCreator = (process: string, id: string): ChangeProcessStatusAction => {
  return {
    type: constants.CHANGE_PROCESS_STATUS,
    process,
    id,
  };
};

export const ChangeTabStatusActionCreator = (status: EnumItemProcessTypes): ChangeTabStatusAction => {
  return {
    type: constants.CHANGE_TAB_STATUS,
    status,
  };
};
