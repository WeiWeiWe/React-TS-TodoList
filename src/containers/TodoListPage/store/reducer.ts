import * as constants from './constants';
import * as actionCreators from './actionCreators';
import { ItemTypes } from '../../../types';

export interface defaultStateTypes {
  inputValue: string;
  list: ItemTypes[];
}

const defaultState: defaultStateTypes = {
  inputValue: '',
  list: [],
};

export default (state = defaultState, action: actionCreators.TodoListActionTypes) => {
  switch (action.type) {
    case constants.CHANGE_INPUT_VALUE:
      return {
        ...state,
        inputValue: action.value,
      };
    case constants.ADD_TODO_LIST:
      return {
        ...state,
        inputValue: '',
        list: [...state.list, action.item],
      };
    case constants.DELETE_TODO_LIST:
      const newList = state.list.filter((item, index) => index !== action.index);
      return {
        ...state,
        list: newList,
      };
    default:
      return state;
  }
};
