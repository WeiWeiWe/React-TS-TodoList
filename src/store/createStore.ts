import { createStore, combineReducers } from 'redux';
import { reducer as todoReducer } from '../containers/TodoListPage/store';

const rootReducer = combineReducers({
  todo: todoReducer,
});

const store = createStore(rootReducer);

export type RootState = ReturnType<typeof store.getState>;

export default store;
