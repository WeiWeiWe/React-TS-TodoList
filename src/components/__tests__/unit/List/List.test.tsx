import React from 'react';
import { Provider } from 'react-redux';
import { render, fireEvent } from '@testing-library/react';
import store from '../../../../store/createStore';
import { List, ListIprops } from '../../../../components/List/List';
import { EnumItemProcessTypes } from '../../../../types';

const testProps: ListIprops = {
  list: [{ id: '0', title: 'Buy pencil', process: EnumItemProcessTypes.ACTIVE, isModify: false }],
  deleteTodoList: jest.fn(),
  switchModifyStatus: jest.fn(),
  InputBlur: jest.fn(),
  changeTitleInputValue: jest.fn(),
  keyUpTitleInputValueDone: jest.fn(),
  inputArrayRef: { current: [{ id: '0', element: document.createElement('input') }] },
  changeProcessStatus: jest.fn(),
  tabStatus: EnumItemProcessTypes.All,
};

const renderComponentWithRedux = (testProps: ListIprops) => {
  return (
    <Provider store={store}>
      <List {...testProps} />
    </Provider>
  );
};

describe('Test List Component', () => {
  test('should render the component to match the snapshot', () => {
    const wrapper = render(renderComponentWithRedux(testProps));
    expect(wrapper).toMatchSnapshot();
  });

  test('should dispaly the prompt tag, if there is no list items', () => {
    const { getByTestId } = render(renderComponentWithRedux({ ...testProps, list: [] }));
    const promptTag = getByTestId(/prompt-tag/) as HTMLLIElement;

    expect(promptTag.innerHTML).toBe('<h3>No Items</h3>');
  });

  test('should dispaly the prompt tag, if there is no active list items', () => {
    const { getByTestId } = render(
      renderComponentWithRedux({
        ...testProps,
        list: [{ id: '0', title: 'Buy pencil', process: EnumItemProcessTypes.DONE, isModify: false }],
        tabStatus: EnumItemProcessTypes.ACTIVE,
      }),
    );
    const promptTag = getByTestId(/prompt-tag/) as HTMLLIElement;

    expect(promptTag.innerHTML).toBe('<h3>No Active Items</h3>');
  });

  test('should dispaly the prompt tag, if there is no done list items', () => {
    const { getByTestId } = render(
      renderComponentWithRedux({
        ...testProps,
        list: [{ id: '0', title: 'Buy pencil', process: EnumItemProcessTypes.ACTIVE, isModify: false }],
        tabStatus: EnumItemProcessTypes.DONE,
      }),
    );
    const promptTag = getByTestId(/prompt-tag/) as HTMLLIElement;

    expect(promptTag.innerHTML).toBe('<h3>No Done Items</h3>');
  });

  test('When user click the delete button, should trigger the deleteTodoList function', () => {
    const { getByTestId } = render(renderComponentWithRedux(testProps));
    const deleteButton = getByTestId(/delete-btn/) as HTMLElement;

    fireEvent.click(deleteButton);
    expect(testProps.deleteTodoList).toHaveBeenCalled();
  });

  test('When user click the modify button, should trigger the switchModifyStatus function', () => {
    const { getByTestId } = render(renderComponentWithRedux(testProps));
    const modifyButton = getByTestId(/modify-btn/) as HTMLElement;

    fireEvent.click(modifyButton);
    expect(testProps.switchModifyStatus).toHaveBeenCalled();
  });

  test('When user modify the input value, should trigger the changeTitleInputValue function', () => {
    const { getByTestId } = render(renderComponentWithRedux(testProps));
    const inputElem = getByTestId(/list-item-input/) as HTMLInputElement;
    const inputValue = 'Buy Pencil';

    fireEvent.change(inputElem, { target: { value: inputValue } });
    expect(testProps.changeTitleInputValue).toHaveBeenCalled();
  });

  test('When user key up the enter keyboard, should trigger the keyUpTitleInputValueDone function', () => {
    const { getByTestId } = render(renderComponentWithRedux(testProps));
    const inputElem = getByTestId(/list-item-input/) as HTMLInputElement;

    fireEvent.keyUp(inputElem, { key: 'Enter', code: 'Enter', charCode: 13 });

    expect(testProps.keyUpTitleInputValueDone).toHaveBeenCalled();
  });

  test('When user blur the input element, should trigger the InputBlur function', () => {
    const { getByTestId } = render(renderComponentWithRedux(testProps));
    const inputElem = getByTestId(/list-item-input/) as HTMLInputElement;

    fireEvent.blur(inputElem);

    expect(testProps.InputBlur).toHaveBeenCalled();
  });

  test('When user click the check box, should trigger the changeProcessStatus function', () => {
    const { getByTestId } = render(renderComponentWithRedux(testProps));
    const checkBox = getByTestId(/list-item-check-box/) as HTMLSpanElement;

    fireEvent.click(checkBox);

    expect(testProps.changeProcessStatus).toHaveBeenCalled();
  });
});
