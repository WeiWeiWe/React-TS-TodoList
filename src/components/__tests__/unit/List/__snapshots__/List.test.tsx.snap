// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test List Component should render the component to match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="sc-bdfBQB kWCrCd"
      >
        <ul
          class="list-fields"
        >
          <li
            class="global-block-fields list-item"
            data-testid="list-item"
          >
            <span
              class="list-item-check-box"
              data-testid="list-item-check-box"
            >
              <span
                class="sc-gsTEea dXjbGb"
                color="rgb(50, 197, 250)"
                height="22px"
                rotate="0"
                width="22px"
              >
                <svg
                  class="check-box-btn"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  
                  <path
                    d="M416 128L192 384l-96-96"
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="32"
                  />
                </svg>
              </span>
            </span>
            <input
              class="list-item-input"
              data-testid="list-item-input"
              disabled=""
              type="text"
              value="Buy pencil"
            />
            <i
              class="list-item-modify-icon"
              data-testid="modify-btn"
            >
              <span
                class="sc-gsTEea daUMzG"
                color="rgb(50, 197, 250)"
                height="30px"
                rotate="0"
                width="35px"
              >
                <svg
                  class="modify-btn"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  
                  <path
                    d="M364.13 125.25L87 403l-23 45 44.99-23 277.76-277.13-22.62-22.62zM420.69 68.69l-22.62 22.62 22.62 22.63 22.62-22.63a16 16 0 000-22.62h0a16 16 0 00-22.62 0z"
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="32"
                  />
                </svg>
              </span>
            </i>
            <i
              class="list-item-delete-icon"
              data-testid="delete-btn"
            >
              <span
                class="sc-gsTEea bKCwHa"
                color="rgba(255, 0, 0, 0.9)"
                height="30px"
                rotate="0"
                width="35px"
              >
                <svg
                  class="delete-btn"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  
                  <path
                    d="M368 368L144 144M368 144L144 368"
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="32"
                  />
                </svg>
              </span>
            </i>
          </li>
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="sc-bdfBQB kWCrCd"
    >
      <ul
        class="list-fields"
      >
        <li
          class="global-block-fields list-item"
          data-testid="list-item"
        >
          <span
            class="list-item-check-box"
            data-testid="list-item-check-box"
          >
            <span
              class="sc-gsTEea dXjbGb"
              color="rgb(50, 197, 250)"
              height="22px"
              rotate="0"
              width="22px"
            >
              <svg
                class="check-box-btn"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                
                <path
                  d="M416 128L192 384l-96-96"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="32"
                />
              </svg>
            </span>
          </span>
          <input
            class="list-item-input"
            data-testid="list-item-input"
            disabled=""
            type="text"
            value="Buy pencil"
          />
          <i
            class="list-item-modify-icon"
            data-testid="modify-btn"
          >
            <span
              class="sc-gsTEea daUMzG"
              color="rgb(50, 197, 250)"
              height="30px"
              rotate="0"
              width="35px"
            >
              <svg
                class="modify-btn"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                
                <path
                  d="M364.13 125.25L87 403l-23 45 44.99-23 277.76-277.13-22.62-22.62zM420.69 68.69l-22.62 22.62 22.62 22.63 22.62-22.63a16 16 0 000-22.62h0a16 16 0 00-22.62 0z"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="32"
                />
              </svg>
            </span>
          </i>
          <i
            class="list-item-delete-icon"
            data-testid="delete-btn"
          >
            <span
              class="sc-gsTEea bKCwHa"
              color="rgba(255, 0, 0, 0.9)"
              height="30px"
              rotate="0"
              width="35px"
            >
              <svg
                class="delete-btn"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                
                <path
                  d="M368 368L144 144M368 144L144 368"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="32"
                />
              </svg>
            </span>
          </i>
        </li>
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
